Formula/g/gossip.rb

```diff
@@ -0,0 +1,23 @@
+class Gossip < Formula
+  desc "Desktop client for Nostr written in Rust"
+  homepage "https://github.com/mikedilger/gossip"
+  url "https://github.com/mikedilger/gossip.git",
+      tag:      "v0.8.0",
+      revision: "c9bb3704e94e105d7cf97de67e89ccf7f01ef69b"
+  license "MIT"
+  head "https://github.com/mikedilger/gossip.git", branch: "master"
+
+  depends_on "rust" => :build
+  depends_on :macos # linux build requires additional runtime dependencies
+
+  def install
+    # required for successful build on intel or linux
+    ENV["RUSTFLAGS"] = "--cfg tokio_unstable"
+
+    system "cargo", "install", *std_cargo_args, "--features", "lang-cjk"
+  end
+
+  test do
+    shell_output("#{bin}/gossip help &>/dev/null", 101)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I will look into this. However the cookbook does not say that. I encourage you to update it as needed.

> We **_want_** tests that don’t require any user input and test the basic functionality of the application. For example foo build-foo input.foo is a good test and (despite their widespread use) **_foo --version and foo --help are bad tests. However, a bad test is better than no test at all._**

Reply:
I came up with a valid test assertion but I am unable to use it due to the same issue that came up before (https://github.com/Homebrew/homebrew-core/pull/144330#issuecomment-1728300279). Running the program through homebrew's test function causes the main thread to panic so it cannot produce the expected output that comes out fine without homebrew/ruby getting in the way.

Since this is a homebrew issue and not an issue with the software itself, there is not much else I can do here.

Reply:
I don't see how software panicking if it runs is a Homebrew issue, especially if 6000 other packages work fine.

Reply:
Resolved in 4238c12f0fb7297fea72edb97a050e81e8ba6610.